src/stan_math_backend/Expression_gen.ml

```diff
@@ -34,9 +34,10 @@ let rec stantype_prim_str = function
   | UArray t -> stantype_prim_str t
   | _ -> "double"
 
-let local_scalar ut = function
-  | DataOnly -> stantype_prim_str ut
-  | AutoDiffable -> "local_scalar_t__"
+let local_scalar ut ad =
+  match (ut, ad) with
+  | _, DataOnly | UInt, AutoDiffable -> stantype_prim_str ut
```

Comment:
What about `UArray UInt, AutoDiffable` etc here? Or do we only care about scalars? (Just want to double check.)

Reply:
Hmm, the way this is used it's intended to be called on a non-UArray, but it's not super obvious. I'll just add the recursive case to be safe.